ADR/ADR023-use-uid-in-urls.md

```diff
@@ -0,0 +1,50 @@
+# ADR23: Use UIDs in URLs to replace form-id
+
+Date: 08/01/24
+
+## Status
+Proposed
+
+## Context
+
+We need to switch away from putting database ids in our URLs. Currently, form urls look something like this:
+`https://submit.forms.service.gov.uk/form/1234/apply-for-an-apple/5678`
+
+Where `1234` is the form-id.
+
+We instead want to change our URLs so that they look either like this: 
+
+`https://submit.forms.service.gov.uk/form/51f5b6f7/apply-for-an-apple/5678`
+
+where the 8 char hex code `51f5b6f7` has replaced the form-id, or like this:
+
+`https://submit.forms.service.gov.uk/form/03998919-6bd1-48f3-a788-b3e3d5c9f822/apply-for-an-apple/5678`
+
+where the uuid `03998919-6bd1-48f3-a788-b3e3d5c9f822` has reaplced the form-id.
+
+We also have to handle existing form URLs, and how we want to maintain them. We can either:
+
+ 1. Give all forms old and new a new random character URL, and redirect from any existing form's old URLs
+ 2. Leave old forms with their old URLs, and only use the random character URLs with newly created forms
+
+### Redirecting from old URLs
+
+We will maintain a list of old URLs mapped to new URLs for live forms that are in use. This will be limited to whatever the current set of live forms is (something like 22 live forms at the moment), and it won't grow. 
+ 
+We'll monitor these redirects to see their usage, and we can phase them out if we see that no one uses them after a certain point. 
+
+The redirect mapping will be 1-to-1, so anyone visiting `form/{current-form-id}/apply-for-an-apple/5678` will redirect to `form/{unique-id}/apply-for-an-apple/5678`. Even if we migrate our data so that forms are given new IDs, the old ID mapping will remain so as not to break bookmarked links.
+
+### Existing forms retain their current URLs
```

Comment:
We discussed this as being sub-optimal in a sense as this will confuse future devs/users.

Reply:
Yep, I'll remove it in favour of the redirect system

Reply:
Ah hang on I've had a chat with Dan actually and I get the idea of it being the minimal change for now and we can convert the IDs in future if we need. Sorry I've been in meetings all day and not had a chance to look at this. Worth reinstating if you still have that change?